src/test/rules/terNoMixedSpacesAndTabsRuleTests.ts

```diff
@@ -0,0 +1,177 @@
+import { RuleTester, Failure, Position } from './ruleTester';
+
+const ruleTester = new RuleTester('ter-no-mixed-spaces-and-tabs');
+
+function expecting(errors: [number, number, string][]): Failure[] {
+  return errors.map((err) => {
+    return {
+      failure: err[2],
+      startPosition: new Position(err[0], err[1]),
+      endPosition: new Position()
+    };
+  });
+}
+
+ruleTester.addTestGroup('allow-unmixed-indentation', 'should pass with default option of unmixed indentation', [
+  `
+ let foo = 1;
+\t\t\tlet bar = 2;
+  if (foo === bar) {
+    console.log('foo equals bar');
+\t\t}`
+]);
+
+ruleTester.addTestGroup('allow-tabs', 'should pass with option of "tabs"', [
+  {
+    code: `
```

Comment:
And use `dedent` in here so that each code snippet can be aligned better. See  https://github.com/buzinas/tslint-eslint-rules/blob/master/src/test/rules/terIndentRuleTests.ts#L2928 as an example.